tests/test_s3urlcache.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+import logging
+import os
+import sys
+import unittest
+
+from dss.storage.validator import scrub_index_data, SizeLimitError
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+from dss import Config, BucketConfig
+from dss.config import IndexSuffix, ESDocType, Replica
+from tests.infra import start_verbose_logging
+from hashlib import sha1
+from dss.storage.validator import S3UrlCache
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+start_verbose_logging()
+
+def setUpModule():
+    IndexSuffix.name = __name__.rsplit('.', 1)[-1]
+
+
+def tearDownModule():
+    IndexSuffix.reset()
+
+
+class TestS3UrlCache(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.replica = Replica.aws
+        Config.set_config(BucketConfig.TEST_FIXTURE)
+        cls.blobstore, _, cls.test_fixture_bucket = Config.get_cloud_specific_handles(cls.replica)
+        Config.set_config(BucketConfig.TEST)
+        _, _, cls.test_bucket = Config.get_cloud_specific_handles(cls.replica)
+        cls.url = "http://www.example.com"
+        cls.url_key = "cache/" + sha1(cls.url.encode("utf-8")).hexdigest()
+        cls.cache = S3UrlCache(logger)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.blobstore.delete(cls.test_bucket, cls.url_key)
+
+    def test_caching(self):
+        self.blobstore.delete(self.test_bucket, self.url_key)
```

Comment:
Delete in `setUp` and in `tearDown` and assert non-existence in test. Tests are partial proofs of correctness. Asserting the non-existence makes the proof stronger.